### PR TITLE
[Keyboard] add bajjak keyboard

### PR DESCRIPTION
**Pull request by garykong**
## Description

This keyboard is a 6x7 array keyboard based on ergodox ez.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bajjak/keymaps/5x6/keymap.c [hunk 1]
```diff
@@ -0,0 +1,212 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/bajjak/keymaps/5x6/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/bajjak/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,218 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/bajjak/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]
Does this have rgb matrix or rgb light support? 
If not, then you may want to remove the code and config for it. 

And you may want to rename the function from ergodox to bajjak

**Comment by garykong** [comment]
Dear @drashna 

1. Does this have rgb matrix or rgb light support?
- I haven't deleted the source code for those who need this feature.

2. Add GPL2 + license in keymap.c

**Comment by drashna** [comment]
> * I haven't deleted the source code for those who need this feature.

Does the hardware physically support it?  If not (especially in the case of the rgb matrix), then it should be removed.

**Comment by garykong** [comment]
> > * I haven't deleted the source code for those who need this feature.
> 
> Does the hardware physically support it? If not (especially in the case of the rgb matrix), then it should be removed.

Removed RGB related code.

**Comment by garykong** [comment]
@drashna Could you please check?

File: keyboards/bajjak/rules.mk [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/bajjak/rules.mk:11]:
```suggestion
```

**Review by drashna — APPROVED** [review]

**Comment by garykong** [comment]
@drashna Please confirm this review.

**Review by drashna — APPROVED** [review]

File: keyboards/bajjak/rules.mk [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+MOUSEKEY_ENABLE  = yes  # Mouse keys
+EXTRAKEY_ENABLE  = yes  # Audio control and System control
+CONSOLE_ENABLE   = no   # Console for debug
+COMMAND_ENABLE   = yes  # Commands for debug and configuration
+CUSTOM_MATRIX    = lite # Custom matrix file for the BAJJAK
+NKRO_ENABLE      = yes  # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+UNICODE_ENABLE   = yes  # Unicode
+SWAP_HANDS_ENABLE= yes  # Allow swapping hands of keyboard
+SLEEP_LED_ENABLE = no
+API_SYSEX_ENABLE = no
```
Comment (fauxpark) [inline keyboards/bajjak/rules.mk:25]:
```suggestion
```

File: keyboards/bajjak/readme.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
+# BAJJAK
+* This keyboard is a 6x7 array keyboard based on Ergodox EZ.
+* __Not compatible with Ergodox EZ.__
+
+# Full-Text of Ergodox EZ
```
Comment (fauxpark) [inline keyboards/bajjak/readme.md:5]: Why is this here?

File: keyboards/bajjak/keymaps/default/readme.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# BAJJAK
+## Changelog
+
+* March 2021:
+  * This keymap is for a 6x7 array keyboard based on bajjak EZ.
+  * __Not compatible with bajjak EZ.__
+
+# Full-Text of Ergodox EZ
```
Comment (fauxpark) [inline keyboards/bajjak/keymaps/default/readme.md:8]: As above.

File: keyboards/bajjak/config.h [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x1209
+#define PRODUCT_ID      0x0002
+#define DEVICE_VER      0x0001
+#define PRODUCT         BAJJAK KEYBOARD
```
Comment (fauxpark) [inline keyboards/bajjak/config.h:30]:
```suggestion
#define PRODUCT         BAJJAK
```

File: keyboards/bajjak/bajjak.h [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+
+#ifndef LED_BRIGHTNESS_LO
+#define LED_BRIGHTNESS_LO       15
+#endif
+#ifndef LED_BRIGHTNESS_HI
+#define LED_BRIGHTNESS_HI       255
+#endif
+
+
+inline void BAJJAK_board_led_on(void)      { DDRD |=  (1<<6); PORTD |=  (1<<6); }
```
Comment (fauxpark) [inline keyboards/bajjak/bajjak.h:58]:
```c
inline void bajjak_board_led_on(void)      { DDRD |=  (1<<6); PORTD |=  (1<<6); }
```
etc.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bajjak/bajjak.h (hunk 8 above)
Comment (garykong) [reply keyboards/bajjak/bajjak.h:58]: I don't understand.

File: keyboards/bajjak/readme.md (hunk 5 above)
Comment (garykong) [reply keyboards/bajjak/readme.md]: I updated the readme.md file.

File: keyboards/bajjak/keymaps/default/readme.md (hunk 6 above)
Comment (garykong) [reply keyboards/bajjak/keymaps/default/readme.md:8]: I updated the readme.md file.

File: keyboards/bajjak/bajjak.h (hunk 8 above)
Comment (fauxpark) [reply keyboards/bajjak/bajjak.h:58]: Change all the function names to lowercase.

Comment (garykong) [reply keyboards/bajjak/bajjak.h]: @fauxpark Changed function names to lowercase.

**Review by garykong — COMMENTED** [review]
@drashna @fauxpark I applied all change requests.

File: keyboards/bajjak/bajjak.h [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        { L04, L14, L24, L34, L44,   L54,   L64   }, \
+        { L05, L15, L25, L35, L45,   KC_NO, L65   }, \
+        { L06, L16, L26, L36, KC_NO, KC_NO, L66   }, \
+        { R00, R10, R20, R30, KC_NO, KC_NO, R60   }, \
+        { R01, R11, R21, R31, R41,   KC_NO, R61   }, \
+        { R02, R12, R22, R32, R42,   R52,   R62   }, \
+        { R03, R13, R23, R33, R43,   R53,   R63   }, \
+        { R04, R14, R24, R34, R44,   R54,   R64   }, \
+        { R05, R15, R25, R35, R45,   R55,   R65   },\
+        { R06, R16, R26, R36, R46,   R56,   KC_NO }                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \
```
Comment (fauxpark) [inline keyboards/bajjak/bajjak.h:170]:
```suggestion
        { R06, R16, R26, R36, R46,   R56,   KC_NO } \
```

File: keyboards/bajjak/bajjak.h [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+                                            L63, L62, L61,  R65, R64, R63                      \
+                        )                                                                      \
+    /* matrix positions */                                                                                                                                                                                                                                                                                                                                                                                                                                                               \
+    {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \
```
Comment (fauxpark) [inline keyboards/bajjak/bajjak.h:156]:
```suggestion
                        ) { \
```

File: keyboards/bajjak/bajjak.c [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+
+    matrix_init_user();
+}
+
+void bajjak_blink_all_leds(void)
+{
+    bajjak_led_all_off();
+    bajjak_led_all_set(LED_BRIGHTNESS_DEFAULT);
+    bajjak_right_led_1_on();
+    _delay_ms(50);
```
Comment (fauxpark) [inline keyboards/bajjak/bajjak.c:79]:
```c
    wait_ms(50);
```

etc.

File: keyboards/bajjak/bajjak.h [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+
+#ifndef LED_BRIGHTNESS_LO
+#define LED_BRIGHTNESS_LO       15
+#endif
+#ifndef LED_BRIGHTNESS_HI
+#define LED_BRIGHTNESS_HI       255
+#endif
+
+
+inline void bajjak_board_led_on(void)      { DDRD |=  (1<<6); PORTD |=  (1<<6); }
```
Comment (fauxpark) [inline keyboards/bajjak/bajjak.h:58]:
```c
inline void bajjak_board_led_on(void)      { setPinOutput(D6); writePinHigh(D6); }
```
etc.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bajjak/bajjak.c (hunk 11 above)
Comment (garykong) [reply keyboards/bajjak/bajjak.c]: @fauxpark changed.

File: keyboards/bajjak/bajjak.h (hunk 12 above)
Comment (garykong) [reply keyboards/bajjak/bajjak.h]: @fauxpark changed.

**Review by drashna — APPROVED** [review]

File: keyboards/bajjak/rules.mk [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no   # Virtual DIP switch configuration
+MOUSEKEY_ENABLE  = yes  # Mouse keys
+EXTRAKEY_ENABLE  = yes  # Audio control and System control
+CONSOLE_ENABLE   = no   # Console for debug
+COMMAND_ENABLE   = yes  # Commands for debug and configuration
+CUSTOM_MATRIX    = lite # Custom matrix file for the BAJJAK
+NKRO_ENABLE      = yes  # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (peepeetee) [inline keyboards/bajjak/rules.mk:21]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/bajjak/rules.mk [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+# If you have Left LEDs (see
+# https://geekhack.org/index.php?topic=22780.msg873819#msg873819 for
+# details), include the following define:
+OPT_DEFS += -DLEFT_LEDS
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no   # Virtual DIP switch configuration
```
Comment (peepeetee) [inline keyboards/bajjak/rules.mk:15]:
```suggestion
BOOTMAGIC_ENABLE = no   # Enable Bootmagic Lite
```

**Review by peepeetee — CHANGES_REQUESTED** [review]

File: keyboards/bajjak/rules.mk [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+# Bootloader selection
+BOOTLOADER = halfkay
+
+# If you have Left LEDs (see
+# https://geekhack.org/index.php?topic=22780.msg873819#msg873819 for
+# details), include the following define:
+OPT_DEFS += -DLEFT_LEDS
+
+# Build Options
+#   comment out to disable the options.
```
Comment (fauxpark) [inline keyboards/bajjak/rules.mk:13]:
```suggestion
#   change yes to no to disable
```

File: keyboards/bajjak/rules.mk [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+SLEEP_LED_ENABLE = no
+
+DEBOUNCE_TYPE = eager_pr
+
+# project specific files
+SRC += matrix.c 
+QUANTUM_LIB_SRC += i2c_master.c
+
+# LAYOUTS = bajjak
+
```
Comment (fauxpark) [inline keyboards/bajjak/rules.mk:33]:
```suggestion
```

File: keyboards/bajjak/rules.mk [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+BOOTMAGIC_ENABLE = no   # Enable Bootmagic Lite
+MOUSEKEY_ENABLE  = yes  # Mouse keys
+EXTRAKEY_ENABLE  = yes  # Audio control and System control
+CONSOLE_ENABLE   = no   # Console for debug
+COMMAND_ENABLE   = yes  # Commands for debug and configuration
+CUSTOM_MATRIX    = lite # Custom matrix file for the BAJJAK
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+UNICODE_ENABLE   = yes  # Unicode
+SWAP_HANDS_ENABLE= yes  # Allow swapping hands of keyboard
+SLEEP_LED_ENABLE = no
```
Comment (fauxpark) [inline keyboards/bajjak/rules.mk:24]:
```suggestion
```

File: keyboards/bajjak/info.json [hunk 18]
```diff
@@ -0,0 +1,33 @@
+{
+    "keyboard_name": "bajjak",
+    "width": 17,
+    "height": 8,
```
Comment (fauxpark) [inline keyboards/bajjak/info.json:4]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bajjak/config.h [hunk 19]
```diff
@@ -0,0 +1,119 @@
...
+
+/* disable print */
+// #define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/bajjak/config.h:118]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by garykong — COMMENTED** [review]
Review completed.

File: keyboards/bajjak/readme.md [hunk 20]
```diff
@@ -0,0 +1,2 @@
+# BAJJAK KEYBOARD
+* This keyboard is a 6x7 array keyboard.
```
Comment (drashna) [inline keyboards/bajjak/readme.md:2]: This needs to be filled out, using the default template. 

https://github.com/qmk/qmk_firmware/blob/e49fa43a7f4025f60ff61ff2fa5015a8fec7ddac/data/templates/keyboard/readme.md

File: keyboards/bajjak/bajjak.h [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+    bool       disable_layer_led   :1;
+  };
+} keyboard_config_t;
+
+extern keyboard_config_t keyboard_config;
+
+/*
+ *   LEFT HAND: LINES 115-122
+ *  RIGHT HAND: LINES 124-131
+ */
```
Comment (noroadsleft) [inline keyboards/bajjak/bajjak.h:143]:
```suggestion
```

File: keyboards/bajjak/info.json [hunk 22]
```diff
@@ -0,0 +1,31 @@
+{
+    "keyboard_name": "bajjak",
+
```
Comment (noroadsleft) [inline keyboards/bajjak/info.json:3]:
```suggestion
    "maintainer": "garykong",
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
[drashna's comment regarding the readme file](https://github.com/qmk/qmk_firmware/pull/12377/files#r896318702) still needs action.

Also, be advised that the `VRSN` will probably not work. `version.h` no longer gets compiled by QMK, so the data the keycode needs to work does not exist in the compiled firmware.

File: keyboards/bajjak/readme.md [hunk 23]
```diff
@@ -0,0 +1,2 @@
+# bajjak
+* This keyboard is a 6x7 array keyboard.
```
Comment (noroadsleft) [inline keyboards/bajjak/readme.md:2]:
```suggestion

This keyboard is a 6x7 array keyboard, similar in design to the Ergodox EZ.

  * Keyboard Maintainer: [Gary Kong](https://github.com/garykong)
* Hardware Supported: bajjak keyboard (ATmega32U4)

Make example for this keyboard (after setting up your build environment):

    make bajjak:default

Flashing example for this keyboard:

    make bajjak:default:flash

## Bootloader

Enter the bootloader in 2 ways:

* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/bajjak/rules.mk [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+#
+BOOTMAGIC_ENABLE = no   # Enable Bootmagic Lite
+MOUSEKEY_ENABLE  = yes  # Mouse keys
+EXTRAKEY_ENABLE  = yes  # Audio control and System control
+CONSOLE_ENABLE   = no   # Console for debug
+COMMAND_ENABLE   = yes  # Commands for debug and configuration
+CUSTOM_MATRIX    = lite # Custom matrix file for the BAJJAK
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+UNICODE_ENABLE   = yes  # Unicode
+SWAP_HANDS_ENABLE= yes  # Allow swapping hands of keyboard
```
Comment (noroadsleft) [inline keyboards/bajjak/rules.mk:23]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output

CUSTOM_MATRIX = lite        # Custom matrix file for the BAJJAK
UNICODE_ENABLE = yes        # Unicode
SWAP_HANDS_ENABLE = yes     # Allow swapping hands of keyboard

# Disable unsupported hardware
BACKLIGHT_SUPPORTED = no
AUDIO_SUPPORTED = no
```

File: keyboards/bajjak/rules.mk [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+
+DEBOUNCE_TYPE = sym_eager_pr
+
+# project specific files
+SRC += matrix.c 
+QUANTUM_LIB_SRC += i2c_master.c
+
+# Disable unsupported hardware
+AUDIO_SUPPORTED = no
+BACKLIGHT_SUPPORTED = no
```
Comment (noroadsleft) [inline keyboards/bajjak/rules.mk:33]:
```suggestion
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/bajjak/readme.md [hunk 26]
```diff
@@ -0,0 +1,23 @@
+# bajjak
+
+This keyboard is a 6x7 array keyboard, similar in design to the Ergodox EZ.
+
+  * Keyboard Maintainer: [Gary Kong](https://github.com/garykong)
```
Comment (noroadsleft) [inline keyboards/bajjak/readme.md:5]: This is my fault; my previous suggestion had a typo.

```suggestion
* Keyboard Maintainer: [Gary Kong](https://github.com/garykong)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]
